## Task #1

$H_0: \mu_1 = \mu_2 = \mu_3$

$H_1: \mu_1 \neq \mu_2 \neq \mu_3$

In [5]:
import numpy as np

In [6]:
fb = np.array([173, 175, 180, 178, 177, 185, 183, 182])
hc = np.array([177, 179, 180, 188, 177, 172, 171, 184, 180])
sh = np.array([172, 173, 169, 177, 166, 180, 178, 177, 172, 166, 170])

k = 3

fb_n = len(fb)
hc_n = len(hc)
sh_n = len(sh)

n = fb_n + hc_n + sh_n

print(f'k={k}, n={n}')

k=3, n=28


In [7]:
fb_mean = np.mean(fb)
fb_mean

179.125

In [11]:
hc_mean = np.mean(hc)
hc_mean

178.66666666666666

In [12]:
sh_mean = np.mean(sh)
sh_mean

172.72727272727272

In [13]:
all_mean = np.mean([fb_mean, hc_mean, sh_mean])
all_mean

176.83964646464645

In [14]:
mean_squared_error = lambda arr, mean: sum((i - mean)**2 for i in arr)

In [15]:
S2 = mean_squared_error(np.concatenate([fb, hc, sh]), all_mean)
S2

834.9093651158043

In [16]:
S2F = sum((i - all_mean)**2 * n for i, n in [(fb_mean, fb_n), (hc_mean, hc_n), (sh_mean, sh_n)])
S2F

257.8525469339864

In [17]:
S2Ost = sum([mean_squared_error(fb, fb_mean), mean_squared_error(hc, hc_mean), mean_squared_error(sh, sh_mean)])
S2Ost

577.0568181818181

In [18]:
print(f'{S2} = {S2F + S2Ost}')

834.9093651158043 = 834.9093651158046


In [19]:
disp_F = S2F / (k - 1)
disp_F

128.9262734669932

In [21]:
disp_Ost = S2Ost / (n - k)
disp_Ost

23.082272727272724

In [22]:
F = disp_F / disp_Ost
F

5.585510360713359

In [24]:
alpha = 0.05

In [26]:
F_t = np.mean([3.40, 3.37])
F_t

3.385

$5.586 > 3.385$ - вида спорта атлета вносит статистически значимый эффект в исследование

In [23]:
from scipy import stats

In [24]:
stats.f_oneway(fb, hc, sh)

F_onewayResult(statistic=5.500053450812596, pvalue=0.010482206918698694)